lib/webpacker/source.rb

```diff
@@ -1,23 +1,30 @@
+require "webpacker/manifest"
+require "webpacker/package_json"
+
 # Translates a logical reference for a pack source into the final
 # path needed in the HTML using generated digests.json manifest.
-
 class Webpacker::Source
+  class SourceError < StandardError; end
+
   def initialize(filename)
     @filename = filename
+    @config ||= Webpacker::PackageJson.new
   end
 
   def path
-    if config[:dev_server_host].present?
-      "#{config[:dev_server_host]}/#{filename}"
+    if Rails.env.development? && config.dev_server[:enabled]
```

Comment:
Be sure to allow overriding the dev_sever[:enabled] option by env value, as this is something that is NOT the application config, but really a runtime option!